plugins/org.python.pydev.parser/src/org/python/pydev/parser/grammarcommon/AbstractTreeBuilder.java

```diff
@@ -833,4 +834,22 @@ protected final SimpleNode makeWithStmt(int arity) {
         return new With(items, s, stack.getGrammar().getInsideAsync());
     }
 
+    public final Assign typedDeclaration(int arity, JJTPythonGrammarState stack, CtxVisitor ctx)
+            throws Exception {
+        exprType type;
+        exprType[] exprs;
+        if (arity >= 3 || stack.nodes.size() >= 3) {
```

Comment:
This is not ok, you are just putting the value in the assign statement again by checking if `stack.nodes.size() >= 3`.

Reply:
Yep, when I made this fix I did not see the .pyi version of the file, so wast not understanding why the message was being typed with 'A' (which is a classdef) if the statement was just something like `self.msg = msg`